### ISSUE TITLE
False positive on no-this-alias

### ISSUE THREAD

**Issue by kitsonk** [denoland/deno_lint#246]
The following code triggers the rule `no-this-alias` inappropriately:

```ts
const { value } = this;
```

The rule should be catching only things like this:

```ts
const self = this;
```
